### PR TITLE
To allow mysql no password local access

### PR DESCRIPTION
**Pull request by wzbbbb**
with or without password, mysql and mysqldump commands use different
syntax.
	modified:   snapshots.py

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/196/

File: cloudferrylib/os/actions/snapshots.py [hunk 1]
```diff
@@ -27,14 +27,18 @@ def run(self, *args, **kwargs):
         # dump mysql to file
         # probably, we have to choose what databases we have to dump
         # by default we dump all databases
+        user = self.cloud.cloud_config.mysql.user
+        password = self.cloud.cloud_config.mysql.password
+        path = self.cloud.cloud_config.snapshot.snapshot_path
+        if password == '':
+            password_arg = ''
+        else:
+            password_arg = "--password=" + password
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/snapshots.py]: Please keep [DRY](http://en.wikipedia.org/wiki/Don't_repeat_yourself) and move the password/no password logic to a separate method/class.

File: cloudferrylib/os/actions/snapshots.py [hunk 2]
```diff
@@ -27,14 +27,18 @@ def run(self, *args, **kwargs):
...
-                   "--user={user} "
-                   "--password={password} "
+                   "--user={0} "
+                   "{1} "
                    "--opt "
-                   "--all-databases > {path}").format(
-            user=self.cloud.cloud_config.mysql.user,
-            password=self.cloud.cloud_config.mysql.password,
-            path=self.cloud.cloud_config.snapshot.snapshot_path)
+                   "--all-databases > {2}").format(user, password_arg, path)
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/snapshots.py]: Changing `{user}` and `{password}` to `{0}` and `{1}` is less readable than it used to be, please revert

File: cloudferrylib/os/actions/snapshots.py [hunk 3]
```diff
@@ -56,13 +60,17 @@ class MysqlRestore(action.Action):
...
         command = ("mysql "
-                   "--user={user} "
-                   "--password={password} "
-                   "< {path}").format(
-            user=self.cloud.cloud_config.mysql.user,
-            password=self.cloud.cloud_config.mysql.password,
-            path=self.cloud.cloud_config.snapshot.snapshot_path)
+                   "--user={0} "
+                   "{1} "
+                   "< {2}").format(user, password_arg, path)
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/snapshots.py]: Same as above

File: cloudferrylib/os/actions/snapshots.py (hunk 1 above)
Comment (roman-verchikov) [inline cloudferrylib/os/actions/snapshots.py]: Also please check if this is the only place with passwordless mysql query not working. If there are more places with the same issue - please fix them too as part of this patch

**Comment by wzbbbb** [comment]
This pull request is not necessary anymore.

**Pull request closed without merging**